src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/StuckEnvelopesService.java

```diff
@@ -0,0 +1,205 @@
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.ZipFileProcessor;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.ZipInputStream;
+
+import static java.time.Instant.now;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static java.util.stream.Collectors.groupingBy;
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.CREATED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_FAILURE;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_SIGNATURE_FAILURE;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_UPLOADED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_UPLOAD_FAILURE;
+
+/**
+ * Service responsible to re-upload stuck envelopes ended in this state after main processor task.
+ * <p></p>
+ * {@link uk.gov.hmcts.reform.bulkscanprocessor.tasks.BlobProcessorTask}
+ * <p></p>
+ * All the validations have been accomplished and {@link Envelope} saved in DB.
+ * Therefore no need to go through the same set of validation steps/rules and just re-extract zip file and re-upload.
+ * Very edge case can cause unrecoverable exception but at current state it will be overridden with
+ * {@link Event#DOC_UPLOAD_FAILURE} which can be treated as stuck envelope again and be re-uploaded.
+ * Review ^ last sentence and, perhaps, incorporate such possibility here.
+ */
+@Service
+public class StuckEnvelopesService {
+
+    private static final Logger log = getLogger(StuckEnvelopesService.class);
+
+    private final long stuckPeriod;
+    private final EnvelopeRepository envelopeRepository;
+    private final BlobManager blobManager;
+    private final ZipFileProcessor zipFileProcessor;
+    private final ProcessEventRepository eventRepository;
+    private final DocumentProcessor documentProcessor;
+    private final EnvelopeProcessor envelopeProcessor;
+
+    public StuckEnvelopesService(
+        @Value("${}") long stuckPeriod,
+        EnvelopeRepository envelopeRepository,
+        BlobManager blobManager,
+        ZipFileProcessor zipFileProcessor,
+        ProcessEventRepository eventRepository,
+        DocumentProcessor documentProcessor,
+        EnvelopeProcessor envelopeProcessor
+    ) {
+        this.stuckPeriod = stuckPeriod;
+        this.envelopeRepository = envelopeRepository;
+        this.blobManager = blobManager;
+        this.zipFileProcessor = zipFileProcessor;
+        this.eventRepository = eventRepository;
+        this.documentProcessor = documentProcessor;
+        this.envelopeProcessor = envelopeProcessor;
+    }
+
+    public void reProcessEnvelopes() {
+        envelopeRepository
+            .findByStatus(CREATED)
+            .stream()
+            // can be moved to query instead. up for a change later or now
```

Comment:
it's only temporary (until upload gets removed from the main job) so I think it's fine to leave it like that

Reply:
yes, realised and updated the comment as well